Assets/Scripts/Scene/SceneController.cs

```diff
@@ -1,165 +1,190 @@
-            Player.characterDNA.UpdateBlock("FEET", modelTextLookup["FEET"], modelColorLookup["FEET"]);
-            Player.characterDNA.UpdateBlock("HAIR", modelTextLookup["HAIR"], modelColorLookup["HAIR"]);
-            Player.characterDNA.UpdateBlock("HANDS", modelTextLookup["HANDS"], modelColorLookup["HANDS"]);
-            Player.characterDNA.UpdateBlock("LEGS", modelTextLookup["LEGS"], modelColorLookup["LEGS"]);
-            Player.characterDNA.UpdateBlock("BODY", modelTextLookup["BODY"], modelColorLookup["BODY"]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Neck, _modelTextLookup[DNABlockType.Neck], _modelColorLookup[DNABlockType.Neck]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Back, _modelTextLookup[DNABlockType.Back], _modelColorLookup[DNABlockType.Back]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Chest, _modelTextLookup[DNABlockType.Chest], _modelColorLookup[DNABlockType.Chest]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Feet, _modelTextLookup[DNABlockType.Feet], _modelColorLookup[DNABlockType.Feet]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Hair, _modelTextLookup[DNABlockType.Hair], _modelColorLookup[DNABlockType.Hair]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Hands, _modelTextLookup[DNABlockType.Hands], _modelColorLookup[DNABlockType.Hands]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Legs, _modelTextLookup[DNABlockType.Legs], _modelColorLookup[DNABlockType.Legs]);
+            Player.CharacterDNA.UpdateBlock(DNABlockType.Body, _modelTextLookup[DNABlockType.Body], _modelColorLookup[DNABlockType.Body]);
         }
 
-        void OnGUI() {
-            if (!animationsLoaded) {
+        private void OnGUI()
+        {
+            if (!_animationsLoaded)
+            {
+                var modelsLoaded = AtlasManager.Instance.ModelsLoaded;
+                var modelsTotal = AtlasManager.Instance.ModelsTotal;
                 // Loading message
-                GUI.Box(new Rect(0, 0, Screen.width, Screen.height), "Caching all sprites... " + 
-                    Math.Floor((AtlasManager.GetModelsLoaded() / AtlasManager.GetModelsTotal() * 100)) + "%");
-                if (AtlasManager.GetModelsLoaded() == AtlasManager.GetModelsTotal()) {
-                    // The sprites are all cached. Lets initialize the scene.
-                    InitializeCharacterUI();
-                    InitializeCharacter();
-                    InitializeDemoDNA();
-                    animationsLoaded = true;
-                    playerController.enabled = true;
-                }
-            } else {
-                int increaseYAmt = 25;
-                int currentY = 35;
-                int currentX = 10;
+                GUI.Box(new Rect(0, 0, Screen.width, Screen.height), "Caching all sprites... " + Math.Floor((double)modelsLoaded / modelsTotal * 100) + "%");
+                if (modelsLoaded == modelsTotal) return;
```

Comment:
Need to inverse the condition here for the scene to load.

Reply:
Really just the issues I mentioned in my comment where I requested changes. That way nothing that was working is breaking =p

If we get a lot more activity on this I can probably get something setup to run unit tests against each pull request. Will be easier to prevent breaking changes.

As far as features to add.. honestly I haven't put much focus into this project since I released the initial version about a year ago. I've been working on a 3D MMORPG thats taken most of my focus for the past 3 years :O (Wrote my own version of DarkRift to handle all of the networking/serialization even).

Honestly, I'd almost want to take a stab at rewriting the roadmap. Really.. with how I've built this plugin, it seems like the logical next step would be to create some sort of inventory system that would allow you to map unique items to a CharacterDNA slot / model / color. Then, when you go to equip an item, it would render the model/color in the appropriate DNA slot. However.. since this plugin was mostly for demonstration (unless we start having a lot of activity and want to turn it into something more), we would probably want an example list of items rather than putting them in a data store.

Reply:
If you're interested, I could rewrite the Roadmap with my latest thoughts.. submit a PR, and you could comment/add features that you would like to see :D

I feel like having some sort of color palette to "color" your race would be cool. It should work similarly to how I've setup a color shader for items. Should be relatively simple but would add a lot of power to those who want to use this plugin.